src/html/fetch-markdown.js

```diff
@@ -97,6 +97,10 @@ async function fetch(
     };
   } catch (e) {
     if (e && e.statusCode && e.statusCode === 404) {
+      if (path.match(/^\/?404\.(md|html)$/)) {
```

Comment:
The idea is that we will first do a regular `bail` with the proper status code (i.e. `400`, `500`, etc.) and try to fetch the corresponding md file (`400.md`, `500.md`, etc.).
If that is missing, we'll end up here again with a path that matches 404, and perform a `safebail` as fallback

Reply:
hmm, I would do this in an/the error handler at the end of the pipe, and not in fetch.